eth/interop.go

```diff
@@ -17,6 +18,21 @@ func (s *Ethereum) CheckAccessList(ctx context.Context, inboxEntries []common.Ha
 	return s.interopRPC.CheckAccessList(ctx, inboxEntries, minSafety, execDesc)
 }
 
+func (s *Ethereum) inferBlockTime(current *types.Header) (uint64, error) {
+	penultimate := s.BlockChain().GetHeaderByHash(current.ParentHash)
+	if penultimate == nil {
+		// We could use a for loop and retry, but this function is used
+		// in the ingress filters, which should fail fast to maintain uptime.
+		if current.Number.Uint64() == 0 {
+			return 0, errors.New("current head is at genesis: penultimate header is nil")
+		} else {
+			return 0, errors.New("a reorg probably occurred: penultimate header is nil")
+		}
```

Comment:
1. you can check if `current.Number` is 0 prior to doing this, avoiding having to `GetHeaderByHash` if you know there won't be anything for the parent.
1. I would avoid speculating the cause of the nil -- just "cannot infer block time, penultimate block is nil` would suffice.

Reply:
> you can check if current.Number is 0 prior to doing this, avoiding having to GetHeaderByHash if you know there won't be anything for the parent.

I considered this, but decided against it since it will be nonzero in virtually all cases. It is cleaner so I'll change it back.

> I would avoid speculating the cause of the nil -- just "cannot infer block time, penultimate block is nil` would suffice.

Good point, will change.